app/controllers/api/v0/admins_controller.rb

```diff
@@ -0,0 +1,167 @@
+# Insights Service Approval APIs
+#
+# APIs to query approval service
+#
+# OpenAPI spec version: 1.0.0
+#
+# Generated by: https://github.com/swagger-api/swagger-codegen.git
+#
+module Api
+  module V0
+    class AdminsController < ApplicationController
+      include UserOperationsMixin
+      include ApproverOperationsMixin
+
+      def add_group
+        group = Group.create!(group_params)
+        json_response(group, :created)
+      end
+
+      def add_stage
+        req = Request.find(params[:request_id])
+        stage = req.stages.create!(stage_params)
+
+        json_response(stage, :created)
+      end
+
+      def add_template
+        template = Template.create!(template_params)
+        json_response(template, :created)
+      end
+
+      def add_workflow
+        workflow = Template.find(params[:template_id]).workflows.create!(workflow_params)
+        json_response(workflow, :created)
+      end
+
+      def fetch_group_by_id
+        group = Group.find(params[:id])
+
+        json_response(group)
+      end
+
+      def fetch_groups
+        groups = Group.all
+
+        json_response(groups)
+      end
+
+      def fetch_requests
+        reqs = Request.filter(params.slice(:requester, :decision, :state))
+
+        json_response(reqs)
+      end
+
+      def fetch_stages
+        stages = Stage.all
+
+        json_response(stages)
+      end
+
+      def fetch_template_by_id
+        template = Template.find(params[:id])
+        json_response(template)
+      end
+
+      def fetch_template_workflows
+        template = Template.find(params[:template_id])
+        json_response(template.workflows)
+      end
+
+      def fetch_templates
+        templates = Template.all
+        json_response(templates)
+      end
+
+      def fetch_workflow_by_id
+        workflow = Workflow.find(params[:id])
+
+        json_response(workflow)
+      end
+
+      def fetch_workflow_requests
+        workflow = Workflow.find(params[:workflow_id])
```

Comment:
@hsong-rh 
Maybe in a different PR we should use params.require(:workflow_id) this is part of the strong parameters which will ensure that the parameter exists and will raise an error if the parameter is missing.  **ActionController::ParameterMissing** will be raised 
http://signalexception.com/blog/2013/07/14/customizing-the-strong-parameters-missing-parameter-response.html
Rails converts this to a 400 HTTP error code

Reply:
@mkanoor Ok, I'll open another PR to use params.require for these find operations.